drivers/espi/espi_ene_kb1200.c

```diff
@@ -0,0 +1,1154 @@
+/*
+ * Copyright (c) 2024 ENE Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ene_kb1200_espi
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+#include <string.h>
+#include <zephyr/drivers/espi.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+
+#include "espi_utils.h"
+#include <reg/espi.h>
+#include <reg/hif.h>
+#include <reg/kbc.h>
+#include <reg/eci.h>
+#include <reg/dbi.h>
+
+LOG_MODULE_REGISTER(espi, CONFIG_ESPI_LOG_LEVEL);
+
+struct espi_kb1200_config {
+	struct espi_regs *base_addr;
+	struct espivw_regs *vw_addr;
+	struct espioob_regs *oob_addr;
+	struct espifa_regs *fa_addr;
+	struct hif_regs *hif_addr;
+	struct kbc_regs *kbc_addr;
+	struct eci_regs *eci_addr;
+	struct dbi_regs *dbi_addr;
+	uintptr_t vwtab_addr;
+};
+
+struct espi_kb1200_data {
+	sys_slist_t callbacks;
+	struct k_sem oob_tx_lock;
+	struct k_sem oob_rx_lock;
+	struct k_sem flash_lock;
+};
+
+struct ene_signal {
+	uint8_t offset; /*HW table register offset*/
+	uint8_t index;  /*VW index*/
+	uint8_t bit;    /*VW data bit*/
+	uint8_t dir;    /*VW direction*/
+};
+
+#ifdef CONFIG_ESPI_VWIRE_CHANNEL
+/* VW signals used in eSPI */
+static const struct ene_signal vw_in[] = {
+	/* index 02h (In) */
+	[ESPI_VWIRE_SIGNAL_SLP_S3] = {ENE_IDX02_OFS, 0x02, BIT(0), ESPI_MASTER_TO_SLAVE},
```

Comment:
Adopt the new eSPI terminology/macros, refer to https://github.com/zephyrproject-rtos/zephyr/pull/68492